### PR TITLE
s3-tests: s3-test task fixes

### PR DESCRIPTION
**Pull request by rakeshgm**
corrected branch name
some of the tests still fail, but the fix should
be in s3-tests repo under ceph project

Signed-off-by: rakeshgm <rakeshgm@redhat.com>

# Description

Please include Automation development guidelines. Source of Test case - New Feature/Regression Test/Close loop of customer BZs

# Checklist:

- [ ] Create a test case in Polarion reviewed and approved.
- [ ] Create a design/automation approach doc. Optional for tests with similar tests already automated.
- [ ] Review the automation design
- [ ] Implement the test script and perform test runs
- [ ] Submit PR for code review and approve
- [ ] Update Polarin Test with Automation script details and update automation fields
- [ ] If automation is part of Close loop, update BZ flag qe-test_coverage “+” and link Polarion test

### CODE REVIEW

**Comment by rakeshgm** [comment]
> Need to pass an ATTR `(-a)` in line 182 in file `tests/rgw/test_s3.py ` in the if condition.

good catch, will fix it now.

**Comment by rakeshgm** [comment]
> > Need to pass an ATTR `(-a)` in line 182 in file `tests/rgw/test_s3.py ` in the if condition.
> 
> good catch, will fix it now.

done, fixed. please check now.

**Comment by rakeshgm** [comment]
rudra checked the PR sanity.

File: tests/rgw/test_s3.py [hunk 1]
```diff
@@ -65,16 +66,16 @@ def setup_s3_tests(client_node, rgw_node, config, build):
...
-    log.info("Running bootstrap")
     if build.startswith('4'):
         client_node.exec_command(cmd="cd s3-tests")
-        client_node.exec_command(cmd="sudo yum install -y python2-virtualenv python2-devel libevent-devel"
-                                 "libffi-devel libxml2-devel libxslt-devel zlib-devel", check_ec=False)
-        client_node.exec_command(cmd="virtualenv --no-site-packages --distribute virtualenv")
-        client_node.exec_command(cmd="./virtualenv/bin/pip install setuptools==32.3.1")
-        client_node.exec_command(cmd="./virtualenv/bin/pip install -r requirements.txt")
-        client_node.exec_command(cmd="./virtualenv/bin/python setup.py develop")
+        client_node.exec_command(cmd="sudo yum install -y python2-virtualenv python2-devel libevent-devel libffi-devel "
```
Comment (sunilkumarn417) [inline tests/rgw/test_s3.py:71]: Suggesting a list to maintain installation packages.

**Review by sunilkumarn417 — CHANGES_REQUESTED** [review]

File: tests/rgw/test_s3.py (hunk 1 above)
Comment (rakeshgm) [reply tests/rgw/test_s3.py]: done.

**Review by sunilkumarn417 — CHANGES_REQUESTED** [review]

File: tests/rgw/test_s3.py [hunk 2]
```diff
@@ -65,16 +66,17 @@ def setup_s3_tests(client_node, rgw_node, config, build):
     repo_url = "https://github.com/ceph/s3-tests.git"
     client_node.exec_command(cmd="git clone -b {branch} {repo_url}".format(branch=branch, repo_url=repo_url))
 
-    log.info("Running bootstrap")
     if build.startswith('4'):
+        pkgs = ["python2-virtualenv", "python2-devel", "libevent-devel", "libffi-devel", "libxml2-devel",
```
Comment (sunilkumarn417) [inline tests/rgw/test_s3.py:70]: would suggest this list to be placed outside if-else block.

File: tests/rgw/test_s3.py [hunk 3]
```diff
@@ -177,9 +178,11 @@ def execute_s3_tests(client_node):
     """
     log.info("Executing s3-tests")
     try:
-        out, err = client_node.exec_command(
-            cmd="cd s3-tests; S3TEST_CONF=config.yaml ./virtualenv/bin/nosetests -v -a '!fails_on_rgw,!lifecycle'",
-            timeout=3600)
+        if build.startswith('4'):
+            cmd = "cd s3-tests; S3TEST_CONF=config.yaml ~/virtualenv/bin/nosetests -v"
```
[the remark this answers is not in the document]
Comment (rakeshgm) [reply tests/rgw/test_s3.py:183]: lifecycle tests fails in  rhcs 3, so we negating those tests i.e these tests will not run.

**Review by sunilkumarn417 — CHANGES_REQUESTED** [review]

File: tests/rgw/test_s3.py (hunk 2 above)
Comment (rakeshgm) [reply tests/rgw/test_s3.py:70]: these pkgs list is needed only for rhcs 4 and not for rhcs 3. but to think of it we need to address for rhcs 5 too. i will check on it.

Comment (rakeshgm) [reply tests/rgw/test_s3.py:70]: i am not sure how it works for `pacific`. we can address this when we see the problem.

File: tests/rgw/test_s3.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (rakeshgm) [reply tests/rgw/test_s3.py]: done.

**Review by sunilkumarn417 — APPROVED** [review]
LGTM

**Pull request merged**